akka-actor/src/main/scala/akka/event/Logging.scala

```diff
@@ -1289,13 +1285,15 @@ trait LoggingAdapter {
     case Logging.WarningLevel ⇒ isWarningEnabled
     case Logging.InfoLevel    ⇒ isInfoEnabled
     case Logging.DebugLevel   ⇒ isDebugEnabled
+    case _                    ⇒ false
   }
 
   final def notifyLog(level: Logging.LogLevel, message: String): Unit = level match {
     case Logging.ErrorLevel   ⇒ if (isErrorEnabled) notifyError(message)
     case Logging.WarningLevel ⇒ if (isWarningEnabled) notifyWarning(message)
     case Logging.InfoLevel    ⇒ if (isInfoEnabled) notifyInfo(message)
     case Logging.DebugLevel   ⇒ if (isDebugEnabled) notifyDebug(message)
+    case unsupported          ⇒ throw new ConfigurationException(s"Log level '$unsupported' in config is not supported.")
```

Comment:
`IllegalArgumentException`

I don't think it has to come from config

s"Unsupported log level [$level]"